tests/test_dagmc_query.py

```diff
@@ -219,3 +200,60 @@ def test_duplicate_calc(function, message):
             if message in str(w[-1].message):
                 test_pass[1] = True
     assert(all(test_pass))
+
+
+def test_coarseness():
+    """Tests the calc coarseness function
+    """
+    three_vols = df.DagmcFile(test_env['three_vols'])
+    vol = three_vols.entityset_ranges['volumes'][0]
+    three_vols_query = dq.DagmcQuery(three_vols, vol)
+    three_vols_query.calc_coarseness()
+    np.testing.assert_almost_equal(
+        list(three_vols_query._surf_data['coarseness']), list(np.full(6, 0.02)))
+
+
+def test_roughness():
+    """Tests the calc roughness function, __calc_average_roughness() function
+    and __calc_tri_roughness() function.
+    """
+    test_pass = np.full(3, False)
+    pyramid = df.DagmcFile(test_env['pyramid'])
+    pyramid_query = dq.DagmcQuery(pyramid)
+    pyramid_query.calc_roughness()
+    
+    gc_top = 2.0*np.pi-4*np.pi/3.0
+    gc_bottom = 2.0*np.pi-2*np.pi/3.0-np.pi/2.0
+    d_bottom = [1.0/np.tan(np.pi/3.0),
+                0.5*(1.0/np.tan(np.pi/3.0)+1.0/np.tan(np.pi/4.0)), 0]
+    lr_bottom = []
+    lr_top = np.abs(gc_top - gc_bottom)
+    lr_bottom.append(np.abs(gc_bottom - \
+                (d_bottom[0]*gc_top+2*d_bottom[1]*gc_bottom) \
+                /(d_bottom[0]+2*d_bottom[1])))
+    lr_bottom.append(np.abs(gc_bottom - \
+                (d_bottom[0]*gc_top+2*d_bottom[1]*gc_bottom+d_bottom[2]*gc_bottom) \
+                /(d_bottom[0]+2*d_bottom[1]+d_bottom[2])))
+    exp = [lr_bottom[0],lr_bottom[0],lr_bottom[1],lr_bottom[1],lr_top]
+    obs = sorted(list(pyramid_query._vert_data['roughness']))
+    test_pass[0] = np.allclose(obs,exp)
+    
+    # test the __calc_average_roughness function
+    side_length = 5.0
+    s_top = 4*(np.sqrt(3)/4*side_length**2)/3.0
+    s_bottom = [(side_length**2/2.0+2*(np.sqrt(3)/4*side_length**2))/3.0, (side_length**2+2*(np.sqrt(3)/4*side_length**2))/3.0]
+    num = lr_top*s_top+ \
+                2*lr_bottom[0]*s_bottom[0] + \
+                2*lr_bottom[1]*s_bottom[1]
+    denom = s_top+sum(s_bottom)*2
+    exp = num/denom
+    obs = pyramid_query._global_averages['roughness_ave']
+    test_pass[1] = np.allclose([obs],[exp])
+
+    #test the __calc_tri_roughness function
+    tri_roughness = [(lr_bottom[0]+lr_bottom[1]+lr_top)/3.0, (lr_bottom[0]*2+lr_bottom[1])/3.0]
+    exp = [tri_roughness[i] for i in [0, 0, 0, 0, 1, 1]]
```

Comment:
this is a clever and convoluted way to do this array filling. I think you could use `np.full()` instead.

Reply:
```
exp = [tri_roughness[0]] * 4 + [tri_roughness[1]] * 2
```